cdap-ui/app/features/hydratorplusplus/controllers/create/toppanel-ctrl.js

```diff
@@ -157,9 +157,124 @@ class HydratorPlusPlusTopPanelCtrl{
   showSettings() {
     this.state.viewSettings = !this.state.viewSettings;
   }
-}
 
-HydratorPlusPlusTopPanelCtrl.$inject = ['$stateParams', 'HydratorPlusPlusConfigStore', 'HydratorPlusPlusConfigActions', '$uibModal', 'HydratorPlusPlusConsoleActions', 'DAGPlusPlusNodesActionsFactory', 'GLOBALS', 'myHelpers', 'HydratorPlusPlusConsoleStore', 'myPipelineExportModalService', '$timeout', '$scope'];
+  // PREVIEW
+  startTimer() {
+    this.previewTimerInterval = this.$interval(() => {
+      let duration = (new Date() - this.previewStartTime) / 1000;
+      duration = duration >= 0 ? duration : 0;
+
+      let minutes = Math.floor(duration / 60);
+      let seconds = Math.floor(duration % 60);
+      seconds = seconds < 10 ? '0' + seconds : seconds;
+      minutes = minutes < 10 ? '0' + minutes : minutes;
+
+      this.displayDuration = {
+        minutes: minutes,
+        seconds: seconds
+      };
+    }, 500);
+  }
+  stopTimer() {
+    this.$interval.cancel(this.previewTimerInterval);
+  }
+
+  runPreview() {
+    let startTime = new Date();
+    this.previewStartTime = startTime;
+    this.startTimer();
```

Comment:
We might want to start the timer once preview has been started in the backend and UI receives a preview ID.